### PR TITLE
Add voting interpolation method as optional argument to SimpleBundleRead method

### PR DESCRIPTION
**Pull request by christophkeller**
This is a simple update to the `MAPL_SimpleBundleRead` method to include the voting interpolation method as an optional input argument (passed down to `MAPL_CFIORead`). I need this for one of my applications. Currently, calling `MAPL_SimpleBundleRead` always reverts to the default interpolation method without option for voting interpolation. This is a zero-diff change (tested with a GEOSgcm run).

### CODE REVIEW

**Comment by tclune** [comment]
@christophkeller You will need to rebase this change on the MAPL `develop` branch.    @mathomp4 or @tclune can assist if you don't know how to do this safely.

**Comment by christophkeller** [comment]
Ah yes, of course! I just changed the base branch to `develop`

File: base/MAPL_SimpleBundleMod.F90 [hunk 1]
```diff
@@ -762,6 +762,7 @@ Function MAPL_SimpleBundleRead (filename, bundle_name, grid, time, verbose, &
     logical, OPTIONAL,           intent(in)    :: verbose
     character(len=*), optional,  intent(IN)    :: only_vars 
     character(len=*), optional,  intent(IN)    :: expid
+    logical,          optional,  intent(IN)    :: voting
```
Comment (tclune) [inline base/MAPL_SimpleBundleMod.F90:765]: New optional arguments are now required to be after an "unusable" argument.  This enforces keyword use for such arguments and ensures backward compatibility as arguments are added.  (MAPL 3 will move the existing optionals to after this as well.)
```suggestion
    class(KeywordEnforcer), optional, intent(in) :: unused
    logical,          optional,  intent(in)    :: voting
```

File: base/MAPL_SimpleBundleMod.F90 [hunk 2]
```diff
@@ -749,7 +749,7 @@ end subroutine MAPL_SimpleBundleDestroy
 !
 
   Function MAPL_SimpleBundleRead (filename, bundle_name, grid, time, verbose, &
-                                  only_vars, expid, rc ) result (self)
+                                  only_vars, expid, voting, rc ) result (self)
```
Comment (tclune) [inline base/MAPL_SimpleBundleMod.F90:752]:
```suggestion
                                  only_vars, expid, voting, rc ) result (self)
         use mapl_KeywordEnforcerMod
```

**Review by tclune — DISMISSED** [review]

**Comment by mathomp4** [comment]
@tclune I think this might be missing something. Maybe an `_UNUSED_DUMMY(unused)` maybe? I see `_UNUSED_DUMMY(unusable)` in places in MAPL...but not every time we have the keyword enforcer. Hmm.

Either way, the CI is not happy:
```
/root/project/GEOSgcm/src/Shared/@MAPL/base/MAPL_SimpleBundleMod.F90(766): error #6451: A dummy argument name is required in this context.   [UNUSED]
    class(KeywordEnforcer), optional, intent(in) :: unused
----------------------------------------------------^
/root/project/GEOSgcm/src/Shared/@MAPL/base/MAPL_SimpleBundleMod.F90(766): error #8223: An entity declared with the CLASS keyword shall be a dummy argument or have the ALLOCATABLE or POINTER attribute.   [UNUSED]
    class(KeywordEnforcer), optional, intent(in) :: unused
----------------------------------------------------^
/root/project/GEOSgcm/src/Shared/@MAPL/base/MAPL_SimpleBundleMod.F90(766): remark #7712: This variable has not been used.   [UNUSED]
    class(KeywordEnforcer), optional, intent(in) :: unused
----------------------------------------------------^
[ 54%] Built target pfio_open_close.x
compilation aborted for /root/project/GEOSgcm/src/Shared/@MAPL/base/MAPL_SimpleBundleMod.F90 (code 1)
make[2]: *** [src/Shared/@MAPL/base/CMakeFiles/MAPL.base.dir/build.make:335: src/Shared/@MAPL/base/CMakeFiles/MAPL.base.dir/MAPL_SimpleBundleMod.F90.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:6676: src/Shared/@MAPL/base/CMakeFiles/MAPL.base.dir/all] Error 2
```

**Comment by tclune** [comment]
Yes - I've been a bit inconsistent with the name over time.    I think I've generally preferred `unusable` of late.  I'll try to fix.

File: base/MAPL_SimpleBundleMod.F90 [hunk 3]
```diff
@@ -762,6 +763,8 @@ Function MAPL_SimpleBundleRead (filename, bundle_name, grid, time, verbose, &
     logical, OPTIONAL,           intent(in)    :: verbose
     character(len=*), optional,  intent(IN)    :: only_vars 
     character(len=*), optional,  intent(IN)    :: expid
+    class(KeywordEnforcer), optional, intent(in) :: unused
```
Comment (tclune) [inline base/MAPL_SimpleBundleMod.F90:766]:
```suggestion
    class(KeywordEnforcer), optional, intent(in) :: unusable
```

File: base/MAPL_SimpleBundleMod.F90 [hunk 4]
```diff
@@ -749,7 +749,8 @@ end subroutine MAPL_SimpleBundleDestroy
 !
 
   Function MAPL_SimpleBundleRead (filename, bundle_name, grid, time, verbose, &
-                                  only_vars, expid, rc ) result (self)
+                                  only_vars, expid, voting, rc ) result (self)
```
Comment (tclune) [inline base/MAPL_SimpleBundleMod.F90:752]:
```suggestion
                                  only_vars, expid, voting, unusable, rc ) result (self)
```
oops missed this earlier.

**Comment by mathomp4** [comment]
This seems to be zero-diff.

**Review by mathomp4 — APPROVED** [review]

**Review by mathomp4 — APPROVED** [review]

**Pull request merged**